tiene relacion con #232

fauno opened the pull request.

fauno committed:
al actualizar un indice todos los subsiguientes se actualizan en cascada

fauno committed:
le tenemos que decir que ya no es temporal

fauno committed:
los indices pasan a ser definitivos manualmente

fauno committed:
informar cuantos indices temporales se van a afectar

fauno committed:
y también que se reindexan facturas

fauno committed:
codigo muerto y dry

mauriciopasquier commented on app/models/indice.rb:
```diff
@@ -9,10 +9,13 @@ class Indice < ActiveRecord::Base
+
+  # Traer todos los indices temporales futuros a este indice y por lo
+  # tanto afectados por sus actualizaciones
+  def temporales_futuros
```
futuros => siguientes, es más claro que son relativos al actual

mauriciopasquier commented on app/models/indice.rb:
```diff
@@ -9,10 +9,13 @@ class Indice < ActiveRecord::Base
-  after_update  :actualizar_cuotas
+  after_update  :indexar_al_ultimo, :actualizar_cuotas
+
+  # Traer todos los indices temporales futuros a este indice y por lo
```
tildes

mauriciopasquier commented on app/models/indice.rb:
```diff
@@ -75,4 +71,16 @@ def actualizar_cuotas
       end
     end
+
+    # cuando se actualiza este indice, todos los indices temporales
```
tildes

mauriciopasquier commented on app/models/indice.rb:
```diff
@@ -9,10 +9,13 @@ class Indice < ActiveRecord::Base
-  # temporal
-  before_update :ahora_es_definitivo
-  after_update  :actualizar_cuotas
+  after_update  :indexar_al_ultimo, :actualizar_cuotas
```
`after_update :indexar_al_ultimo, unless: :temporal?`

mauriciopasquier commented on test/models/cuota/cuota_con_contrato_test.rb:
```diff
@@ -94,8 +94,7 @@ class CuotaConContratoTest < ActiveSupport::TestCase
 
     # al cambiar el valor del indice, se dispara la actualizacion de
```
tildes

mauriciopasquier commented on test/models/cuota/cuota_con_contrato_test.rb:
```diff
@@ -94,8 +94,7 @@ class CuotaConContratoTest < ActiveSupport::TestCase
     # montos de facturas por Indice.after_update
-    c.indice.valor = c.indice.valor * 2
-    assert c.indice.save
+    assert c.indice.update(valor: c.indice.valor * 2, temporal: false)
```
hace falta `assert`ar el update?

fauno committed:
mas que cascada era catarata

fauno committed:
no assertear el update

fauno commented:
listo, ahora solo se actualizan los indices temporales subsiguientes hasta que se encuentra el siguiente definitivo

fauno committed:
un montón de tildes

fauno committed:
tildes en índices closes #397

m4tu5 commented on app/controllers/indices_controller.rb:
```diff
@@ -21,7 +21,9 @@ def new
-    @editar=true
+    @editar = true
+
+    flash[:notice] = "Al modificar este índice se afectarán #{@indice.temporales_siguientes.count} índices temporales y se reindexarán sus facturas si las tuviesen."
```
se podrá especificar el número de facturas?

no me queda claro por que comentamos esto, porque se va a ver afectado el saldo de las facturas, porque puede tardar mucho tiempo en hacer el proceso?

creo que un mensaje más específico puede ser mejor para el usuario

m4tu5 replied:
veo que `temporales_siguientes` es algo diferente de lo que realmente se actualiza (que solo toma los indices_siguientes hasta el próximo no-temporal).

fauno replied:
tus dos comentarios se cancelan mutuamente

fauno replied:
ah lo borraste. son los siguientes hasta el proximo definitivo, se te ocurre un nombre mejor?

fauno committed:
Retencion.pagar_con siempre devuelve un movimiento

aunque ahora falla la refutacion

BorrarRecibosTest#test_borra_sus_movimientos_a_menos_que_alguno_tenga_causa_trackeable
[/home/fauno/Proyectos/lainventoria/cp/test/models/recibo/b
orrar_recibos_test.rb:20]:
Failed refutation, no message given

fauno committed:
a veces hay contratos sin cuotas

fauno commented:
estaba cerrando errores en los tests y no puedo descular 8afec0c, por qué al crear una retención ya está aplicada? a resolver en #281

mauriciopasquier commented:
estás queriendo deshacer lo que @sudoaza hizo en #385 ?

fauno commented:
si lo hizo @sudoaza que arregle el test el!

fauno commented:
no hay un +1 para mi?

m4tu5 commented:
Aún está pendiente el comentario del mensaje que se muestra al actualizar un indice.

No me parece tan importante cuantos indices posteriores se actualizan (lo que muestra ahora), sino el mensaje de que actualizar el indice va a cambiar el saldo de algunas facturas

fauno committed:
creo que odio a formatted_number

no, es noble y está bien programado, buenas noches

fauno committed:
mostrar la cantidad de facturas que se van a reindexar

fauno commented:
listo

m4tu5 commented:
+1
